app/uk/gov/hmrc/customs/declaration/services/CommunicationService.scala

```diff
@@ -47,18 +47,17 @@ class CommunicationService @Inject()(logger: DeclarationsLogger,
     Future.successful(configuration.getString("override.clientID"))
   }
 
-  def prepareAndSend(inboundXml: NodeSeq, requestedApiVersion: RequestedVersion)(implicit hc: HeaderCarrier): Future[Ids] = {
-    val conversationId = uuidService.uuid()
+  def prepareAndSend(inboundXml: NodeSeq, ids: Ids)(implicit hc: HeaderCarrier): Future[Ids] = {
     val correlationId = uuidService.uuid()
     val dateTime = dateTimeProvider.nowUtc()
 
-    logger.info(s"[conversationId=$conversationId] correlationId=$correlationId, dateTime=$dateTime, requestedApiVersion=$requestedApiVersion")
+    val version = ids.maybeRequestedVersion.getOrElse(throw new IllegalStateException("api version should have been defined by now"))
```

Comment:
This is a smell here, version should never be empty at this stage and we should not be passing on an option here instead it should be v1 or v2.